SU2_DEF/src/drivers/CDeformationDriver.cpp

```diff
@@ -0,0 +1,804 @@
+
+using namespace std;
+
+CDeformationDriver::CDeformationDriver(char* confFile, SU2_Comm MPICommunicator) {
+
+  /*--- Initialize Medipack (must also be here so it is initialized from python) ---*/
+  #ifdef HAVE_MPI
+    #if defined(CODI_REVERSE_TYPE) || defined(CODI_FORWARD_TYPE)
+      SU2_MPI::Init_AMPI();
+    #endif
+  #endif
+
+  SU2_MPI::SetComm(MPICommunicator);
+
+  rank = SU2_MPI::GetRank();
+  size = SU2_MPI::GetSize();
+
+  /*--- Copy the config filename ---*/
+  strcpy(config_file_name, confFile);
+
+  /*--- Initialize the configuration of the driver ---*/
+  driver_config = nullptr;
+  driver_config = new CConfig(config_file_name, SU2_COMPONENT::SU2_DEF);
+
+  nZone = driver_config->GetnZone();
+
+  /*--- Initialize containers --- */
+
+  SetContainers_Null();
+
+  /*--- Preprocessing of the config files. ---*/
+
+  Input_Preprocessing();
+
+  /*--- Set up a timer for performance benchmarking ---*/
+
+  StartTime = SU2_MPI::Wtime();
+
+  /*--- Preprocessing of the geometry for all zones. ---*/
+
+  Geometrical_Preprocessing();
+
+  /*--- Preprocessing of the output for all zones. ---*/
+
+  Output_Preprocessing();
+
+  if (driver_config->GetDeform_Mesh()){
+
+    /*--- Preprocessing of the mesh solver for all zones. ---*/
+
+    Solver_Preprocessing();
+
+    /*--- Preprocessing of the mesh solver for all zones. ---*/
+
+    Numerics_Preprocessing();
+
+  }
+
+  /*--- Preprocessing time is reported now, but not included in the next compute portion. ---*/
+
+  StopTime = SU2_MPI::Wtime();
+
+  /*--- Compute/print the total time for performance benchmarking. ---*/
+
+  UsedTime = StopTime-StartTime;
+  UsedTimePreproc    = UsedTime;
+  UsedTimeCompute    = 0.0;
+
+}
+
+CDeformationDriver::~CDeformationDriver(void) {
+
+}
+
+void CDeformationDriver::SetContainers_Null() {
+
+    /*--- Create pointers to all of the classes that may be used throughout
+        the SU2_DEF code. In general, the pointers are instantiated down a
+        hierarchy over all zones as described in the comments below. ---*/
+    config_container   = new CConfig*[nZone];
+    output_container   = new COutput*[nZone];
+    geometry_container = new CGeometry*[nZone];
+    surface_movement   = new CSurfaceMovement*[nZone];
+    grid_movement      = new CVolumetricMovement*[nZone];
+
+    solver_container   = new CSolver*[nZone];
+    numerics_container = new CNumerics**[nZone];
+
+    for (iZone = 0; iZone < nZone; iZone++) {
+        config_container[iZone]       = nullptr;
+        output_container[iZone]       = nullptr;
+        geometry_container[iZone]     = nullptr;
+        surface_movement[iZone]       = nullptr;
+        grid_movement[iZone]          = nullptr;
+        solver_container[iZone]       = nullptr;
+        numerics_container[iZone]     = nullptr;
+    }
+}
+
+void CDeformationDriver::Input_Preprocessing() {
+
+    /*--- Initialize a char to store the zone filename ---*/
+    char zone_file_name[MAX_STRING_SIZE];
+
```

Comment:
Run clang-format on the new files at some point please. We have been doing that for new files to make everything consistent.
The format file is in the root of the repo.